src/Test/WordPress/ProxyTest.php

```diff
@@ -2,78 +2,143 @@
+        $mockWPVerifyNonce->expects($this->once())->willReturn(true);
+        $this->mockRequestRouter->expects($this->once())->method('route');
+        $mockWPDie = $this->getFunctionMock('CF\WordPress', 'wp_die');
+        $this->proxy->run();
+    }
+
+    public function testIsCloudFlareCSRFTokenValidReturnsTrueForGet()
+    {
+        $this->assertTrue($this->proxy->isCloudFlareCSRFTokenValid('GET', null));
+    }
+
+    public function testCreateRequestGETProxyClient()
+    {
+        $url = 'testproxyurl.com';
+        $proxyURL = 'proxyurl.com';
+        $_SERVER['REQUEST_METHOD'] = 'GET';
+        $_GET['proxyURL'] = $url;
+        $_GET['proxyURLType'] = 'CLIENT';
+        $jsonBody = json_encode(array(
+            'proxyURL' => $proxyURL,
+        ));
+
+        $mockFileGetContents = $this->getFunctionMock('CF\WordPress', 'file_get_contents');
+        $mockFileGetContents->expects($this->any())->willReturn($jsonBody);
+
+        $request = $this->proxy->createRequest();
 
+        $this->assertFalse(isset($request->getParameters()['proxyURL']));
+        $this->assertFalse(isset($request->getParameters()['proxyURLType']));
+        $this->assertFalse(isset($request->getBody()['proxyURL']));
+        $this->assertEquals(\CF\API\Client::ENDPOINT.$url, $request->getUrl());
+    }
+
+    public function testCreateRequestGETProxyPlugin()
+    {
+        $url = 'testproxyurl.com';
+        $proxyURL = 'proxyurl.com';
+        $_SERVER['REQUEST_METHOD'] = 'GET';
+        $_GET['proxyURL'] = $url;
+        $_GET['proxyURLType'] = 'PLUGIN';
+        $jsonBody = json_encode(array(
+            'proxyURL' => $proxyURL,
+        ));
+
+        $mockFileGetContents = $this->getFunctionMock('CF\WordPress', 'file_get_contents');
+        $mockFileGetContents->expects($this->any())->willReturn($jsonBody);
+
+        $request = $this->proxy->createRequest();
+
+        $this->assertFalse(isset($request->getParameters()['proxyURL']));
+        $this->assertFalse(isset($request->getParameters()['proxyURLType']));
+        $this->assertFalse(isset($request->getBody()['proxyURL']));
+        $this->assertEquals(\CF\API\Plugin::ENDPOINT.$url, $request->getUrl());
+    }
+
+    public function testCreateRequestNonGET()
+    {
+        $proxyURL = 'proxyurl.com';
+        $_SERVER['REQUEST_METHOD'] = 'POST';
+        $jsonBody = json_encode(array(
+            'proxyURL' => $proxyURL,
+        ));
+
+        $mockFileGetContents = $this->getFunctionMock('CF\WordPress', 'file_get_contents');
+        $mockFileGetContents->expects($this->any())->willReturn($jsonBody);
+
+        $request = $this->proxy->createRequest();
+
+        $this->assertFalse(isset($request->getParameters()['proxyURL']));
+        $this->assertFalse(isset($request->getParameters()['proxyURLType']));
+        $this->assertFalse(isset($request->getBody()['proxyURL']));
+        // $this->assertEquals($proxyURL, $request->getUrl());
```

Comment:
Instead of the method you suggested with `protected $jsonBody` and set the json in the contructor I took this approach for `file_get_content` because if we map the getting input logic to constructor then it'd have restricted our ability to move the constructor. 

Also I'm not even certain if it'd have worked in the current situation because we need to get the input when the hook calls `$this->proxy->run();` not before.